feat(HTTP): Refresh 3XX response pages (#35353)
mdn/content: files/en-us/web/http/status/300/index.md

{{HTTPSidebar}}

The HTTP **`300 Multiple Choices`** [redirection response](/en-US/docs/Web/HTTP/Status#redirection_messages) status code indicates that the request has more than one possible response.
The user-agent or the user should choose one of them.

> There is no standardized way of choosing one of the responses, so this response code is rarely used.

Reviewer: The index above this states " (There is no standardized way of choosing one of the responses, but HTML links to the possibilities are recommended so the user can pick.)"

If that is true, we should mention it. It would be true if it says so in the spec.

Author: spec says:

>A specific format for automatic selection is not defined by this specification because HTTP tries to remain orthogonal to the definition of its content. In practice, the representation is provided in some easily parsed format believed to be acceptable to the user agent, as determined by shared design or content negotiation, or in some commonly accepted hypertext format.

Likely HTML or JSON. Edit incoming, happy to hear what you think

Author: changed in 2771552629bf8ba63be74b5f5830ea93982707ef

Reviewer: This is good thanks.

If the server has a preferred choice that the client should request, it should generate a {{HTTPHeader("Location")}} header.
See [Content negotiation](/en-US/docs/Web/HTTP/Content_negotiation) for more information.

## Status

```http
300 Multiple Choices
```

## Examples

### 300 response with a `Location` header

> The following example shows how reactive negotiation (or [agent-driven negotiation](/en-US/docs/Web/HTTP/Content_negotiation#agent-driven_negotiation)) may look like.

Reviewer: ```suggestion
The following example shows how reactive negotiation (or [agent-driven negotiation](/en-US/docs/Web/HTTP/Content_negotiation#agent-driven_negotiation)) may look.
```

> In this scenario, a `300` response with details of different representations of a resource is sent by the server, and the client chooses which resource it wants.
> The downsides to this approach is that the content may have a different format or syntax depending on the implementation, and that it includes an additional request-response exchange which slows down the overall client-server interaction:

Reviewer: ```suggestion
In this scenario, a `300` response is sent by the server with details of different representations of the requested resource, and the client chooses which resource it wants.

The downsides to this approach are that the content may have a different format or syntax depending on the implementation, and that the interaction includes an additional request-response exchange — which slows down the interaction:
```

Reviewer: While this is OK, I think I prefer that we clearly identify the request and response, so also proposing another way of doing this below.

```http
GET /resource HTTP/1.1
Host: example.com
Accept: application/json
```

```http
HTTP/1.1 300 Multiple Choices
Content-Type: application/json
Location: http://example.com/resource-html

{
  "message": "Multiple choices available.",
  "choices": [
    {
      "contentType": "text/pdf",
      "uri": "http://example.com/resource-pdf"
    },
    {
      "contentType": "text/html",
      "uri": "http://example.com/resource-html"
    }
  ]
}
```

> 

Reviewer: ```suggestion

The downsides of this approach are that the content may have a different format or syntax depending on the implementation, and that the interaction includes an additional request-response exchange — which slows down the interaction.

```
